provider/lib/home/home_screen.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:flutter/material.dart' hide Action;
+import 'package:provider/provider.dart';
+import 'package:provider_sample/home/stats_view.dart';
+import 'package:provider_sample/home/todo_list_view.dart';
+import 'package:provider_sample/localization.dart';
+import 'package:provider_sample/todo_list_model.dart';
+import 'package:todos_app_core/todos_app_core.dart';
+
+import '../models.dart';
+import 'extra_actions_button.dart';
+import 'filter_button.dart';
+
+class HomeScreen extends StatefulWidget {
+  const HomeScreen();
+
+  @override
+  _HomeScreenState createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  // Because the state of the tabs is only a concern to the HomeScreen Widget,
+  // it is stored as local state rather than in the TodoListModel.
+  final _tab = ValueNotifier(_HomeScreenTab.todos);
+
+  @override
+  void dispose() {
+    _tab.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(ProviderLocalizations.of(context).appTitle),
+        actions: <Widget>[
+          AnimatedBuilder(
+            animation: _tab,
+            builder: (_, __) => FilterButton(
+              isActive: _tab.value == _HomeScreenTab.todos,
+            ),
+          ),
+          const ExtraActionsButton(),
+        ],
+      ),
+      floatingActionButton: FloatingActionButton(
+        key: ArchSampleKeys.addTodoFab,
+        onPressed: () => Navigator.pushNamed(context, ArchSampleRoutes.addTodo),
+        tooltip: ArchSampleLocalizations.of(context).addTodo,
+        child: const Icon(Icons.add),
+      ),
+      body: Selector<TodoListModel, bool>(
+        selector: (context, model) => model.isLoading,
+        builder: (context, isLoading, _) {
+          if (isLoading) {
+            return Center(
+              child: CircularProgressIndicator(
+                key: ArchSampleKeys.todosLoading,
+              ),
+            );
+          }
+
+          return AnimatedBuilder(
```

Comment:
Can be moved to `child`:

```dart
Selector(
  ...
  builder: (_, isLoading, child) {
    if (isLoading) return Something();
   return child;
  },
  child: AnimatedBuilder(...),
)
```

Reply:
Also, we have [ValueListenableBuilder](https://api.flutter.dev/flutter/widgets/ValueListenableBuilder-class.html) now.